bin/node/src/flags/p2p.rs

```diff
@@ -36,13 +36,35 @@ pub struct P2PArgs {
     /// The hex-encoded 32-byte private key for the peer ID.
     #[arg(long = "p2p.priv.raw", env = "KONA_NODE_P2P_PRIV_RAW")]
     pub private_key: Option<B256>,
-    /// IP to bind LibP2P and Discv5 to.
+
+    /// IP to advertise to external peers from Discv5.
+    /// Optional argument. Use the `p2p.listen.ip` if not set.
+    #[arg(long = "p2p.advertise.ip", env = "KONA_NODE_P2P_ADVERTISE_IP")]
+    pub advertise_ip: Option<IpAddr>,
+    /// TCP port to advertise to external peers from the discovery layer. Same as `p2p.listen.tcp`
+    /// if set to zero.
+    #[arg(
+        long = "p2p.advertise.tcp",
+        default_value = "0",
+        env = "KONA_NODE_P2P_ADVERTISE_TCP_PORT"
+    )]
+    pub advertise_tcp_port: u16,
```

Comment:
Only note that the approach to use `0` as a way to re-use the `p2p.listen.tcp` port prevents people from using `0` as a way to grab an ephemeral/dynamically assigned port. e.g. there may be the edge case here where the user doesn't want the advertised ports to be the same as the listening ports, but also wants to use `0` in order to get a dynamically assigned port. Probably not a big deal, but worth mentioning.